pkg/gadgets/bpftrace/tracer.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !withoutebpf
+
+package tracer
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/exec"
+
+	gadgetcontext "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-context"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/bpftrace/types"
+)
+
+type Config struct {
+	Program string
+}
+
+type Tracer struct {
+	eventCallback func(ev *types.Event)
+	config        Config
+}
+
+func (t *Tracer) Run(gadgetCtx gadgets.GadgetContext) error {
+	params := gadgetCtx.GadgetParams()
+	t.config.Program = params.Get(ParamProgram).AsString()
```

Comment:
Shouldn't it check that the string is not empty?

Reply:
I handled it using a validator in gadget.go. We're still discussing how to perform server-side validation of parameters.

Reply:
Server side validation is handled in `service.go` for the grpc way (`.CopyFromMap()`). So if not called manually (or through the current CR way), it's should have already been taken care of.

Reply:
right, I opened https://github.com/inspektor-gadget/inspektor-gadget/pull/1486 to keep gadgets aligned.